xLib/src/main/java/java10/xLib/PolynomialExpression.java

```diff
@@ -0,0 +1,44 @@
+package java10.xLib;
+
+public abstract class PolynomialExpression implements Expression {
+
+	protected Expression a;
+	protected Expression b;
+
+	public PolynomialExpression(Expression a, Expression b) {
+		this.a = a;
+		this.b = b;
+	}
+
+	public PolynomialExpression() {
+	}
+
+	public Expression getA() {
+		return a;
+	}
+
+	public void setA(Expression a) {
+		this.a = a;
+	}
+
+	public Expression getB() {
+		return b;
+	}
+
+	public void setB(Expression b) {
+		this.b = b;
+	}
+
+	@Override
+	public abstract double evaluate(double x);
```

Comment:
You can define new abstract function, such as doEvaluate(double first, double second) and override at child classes.

 ```suggestion
	public double evaluate(double x) {
	    return doEvaluate(a.evaluate(x), b.evaluate(y));
	}
```